editor/src/components/canvas/canvas.ts

```diff
@@ -46,84 +126,7 @@ const Canvas = {
-
-    function recurseChildren(componentTree: ElementPathTree): {
-      boundingRect: CanvasRectangle | null
-      frames: Array<FrameWithPath>
-    } {
-      const component = MetadataUtils.findElementByElementPath(metadata, componentTree.path)
-      if (component == null) {
-        return {
-          boundingRect: null,
-          frames: [],
-        }
-      }
-      const globalFrame = component.globalFrame
-      if (globalFrame == null) {
-        return {
-          boundingRect: null,
-          frames: [],
-        }
-      }
-
-      const overflows = MetadataUtils.overflows(allElementProps, componentTree.path)
-      const includeClippedNext = useBoundingFrames && overflows
-
-      let children: Array<ElementPathTree> = []
-      let unfurledComponents: Array<ElementPathTree> = []
-      fastForEach(componentTree.children, (childTree) => {
-        if (EP.isRootElementOfInstance(childTree.path)) {
-          unfurledComponents.push(childTree)
-        } else {
-          children.push(childTree)
-        }
-      })
-
-      const childFrames = children.map((child) => {
-        const recurseResults = recurseChildren(child)
-        const rectToBoundWith = includeClippedNext ? recurseResults.boundingRect : globalFrame
-        return { boundingRect: rectToBoundWith, frames: recurseResults.frames }
-      })
-      const unfurledFrames = unfurledComponents.map((unfurledElement) => {
-        const recurseResults = recurseChildren(unfurledElement)
-        const rectToBoundWith = includeClippedNext ? recurseResults.boundingRect : globalFrame
-        return { boundingRect: rectToBoundWith, frames: recurseResults.frames }
-      })
-      const allFrames = [...childFrames, ...unfurledFrames]
-      const allChildrenBounds = Utils.boundingRectangleArray(Utils.pluck(allFrames, 'boundingRect'))
-      if (allFrames.length > 0 && allChildrenBounds != null) {
-        const allChildrenFrames = Utils.pluck(allFrames, 'frames').flat()
-        const boundingRect = Utils.boundingRectangle(globalFrame, allChildrenBounds)
-        const toAppend: FrameWithPath = { path: component.elementPath, frame: boundingRect }
-        return {
-          boundingRect: boundingRect,
-          frames: [toAppend].concat(allChildrenFrames),
-        }
-      } else {
-        const boundingRect = globalFrame
-        const toAppend = { path: component.elementPath, frame: boundingRect }
-        return { boundingRect: boundingRect, frames: [toAppend] }
-      }
-    }
-
-    const storyboardChildren = MetadataUtils.getAllStoryboardChildrenPaths(metadata)
-    return storyboardChildren.flatMap((storyboardChild) => {
-      const subTree = getSubTree(projectTree, storyboardChild)
-      if (subTree == null) {
-        return []
-      } else {
-        return recurseChildren(subTree).frames
-      }
-    })
-  },
+  getFramesInCanvasContext: memoize(getFramesInCanvasContextUncached, { maxSize: 1 }),
```

Comment:
maybe it's worth making the maxSize 2 or more, because I can imagine some scenario where we call it with the unpatched editor and the patched editor state within the same frame

Reply:
Yeah, I put 1 in and it made a bunch of calls disappear, but it's probably worth giving it that bit of extra wiggle room.

Changed in c20b2a94c.